File: examples/sequential-cpp/example2.cpp
/* This code calculates an arctan approximation for |x| < 1 */
#include <iostream>
#include <cmath>
#include "utils/cppheader.h"

const int LIMIT = 1000000000;

using namespace std;

class Atan {
private:
	double myX, result;
	
public:
	Atan(double x) : myX(x), result(0) {}
	
	double getResult() const {
		return result;
	}
	
	void calculate() {
		double one, n;
		
		result = 0;
		for (int j = 0; j < LIMIT; j++) {
			one = (j % 2 == 0)? 1.0 : -1.0;
			n = (2 * j ) + 1;
			result += ( (one / n) * pow(myX, n) );
		}
	}
};

int main(int argc, char* argv[]) {
	double ms;
	Timer t;
	
	ms = 0;
	Atan obj(0.99);
	cout << "Starting..." << endl;
	for (int i = 0; i < N; i++) {
		t.start();
		obj.calculate();
		ms += t.stop();
	}
	cout << "arctan(0.99)->(0.78) = " << obj.getResult() << endl;
	cout << "avg time = " << (ms / N) << " ms" << endl;
	
	return 0;
}

File: examples/sequential-java/Example2.java
/* This code calculates de atan of x, x < 1 */

public class Example2 {
	private static final int LIMIT = 100_000_000;
	private double result, x;
	
	public Example2(double x) {
		this.x = x;
		this.result = 0;
	}
	
	public double getResult() {
		return result;
	}
	
	public void calculate() {
		double one, n;
		
		result = 0;
		for (int j = 0; j < LIMIT; j++) {
			one = (j % 2 == 0)? 1.0 : -1.0;
			n = (2 * j ) + 1;
			result += ( (one / n) * Math.pow(x, n) );
		}
	}
	
	public static void main(String args[]) {
		long startTime, stopTime;
		double acum = 0;
		
		Example2 e = new Example2(0.99);
		acum = 0;
		for (int i = 0; i < Utils.N; i++) {
			startTime = System.currentTimeMillis();
			e.calculate();
			stopTime = System.currentTimeMillis();
			acum += (stopTime - startTime);
		}
		System.out.printf("arctan(0.99)->(0.78) = %f\n", e.getResult());
		System.out.printf("avg time = %.5f\n", (acum / Utils.N));
	}
}
			
